src/libostree/ostree-repo-pull.c

```diff
@@ -4027,12 +4027,44 @@ ostree_repo_pull_with_options (OstreeRepo             *self,
     if (pull_data->gpg_verify_summary && bytes_summary && bytes_sig)
       {
         g_autoptr(OstreeGpgVerifyResult) result = NULL;
+        g_autoptr(GError) temp_error = NULL;
 
         result = ostree_repo_verify_summary (self, pull_data->remote_name,
                                              bytes_summary, bytes_sig,
                                              cancellable, error);
-        if (!ostree_gpg_verify_result_require_valid_signature (result, error))
-          goto out;
+        if (!ostree_gpg_verify_result_require_valid_signature (result, &temp_error))
+          {
+            if (summary_from_cache)
+              {
+                /* The cached summary doesn't match, fetch a new one and verify again */
+                summary_from_cache = FALSE;
+                g_clear_pointer (&bytes_summary, (GDestroyNotify)g_bytes_unref);
+                g_warning ("Remote %s cached summary invalid, pulling new version",
```

Comment:
See comment in https://github.com/ostreedev/ostree/pull/1685#discussion_r204052444

Reply:
I added it because it should really never happen anymore (and the test suite is run with `G_DEBUG=fatal-warnings`). The idea is that if we do start seeing a constant stream of warnings containing this then a new bug has been introduced in ostree that we probably want to fix instead of silently ignoring the summary cache all the time and making it useless.

That said, it is your project so I can drop it or change it to `g_debug`.

Reply:
I generally think the choice should be between `GError**` and `g_assert_not_reached()`. Since the control flow here is affected by external data, `g_assert_not_reached()` is not a good idea, so this should probably be a `GError**`.

Reply:
@pwithnall but in this case it's not an error since it's being recovered from. A returned `GError` would likely never be seen.

I'll just change it to a `g_debug`. Hopefully the test suite will catch if new errors are introduced in the summary cache handling.